manifests/gid.pp

```diff
@@ -0,0 +1,20 @@
+#
+# Provides system-wide defaults for GID settings
+#
+# @param min
+#   The lowest allowed regular user GID for the system
+#
+# @param max
+#   The highest allowed regular user GID for the system
+#
+#   * If not defined, applications should simply do what makes sense for them
+#     internally
+#
+# @author SIMP Team - https://simp-project.com
+#
+class simp_options::gid (
+  Integer[0]           $min = pick(fact('login_defs.gid_min'), 1000),
```

Comment:
Why the pick()?  The new fact sets a default value for gid_min if it is not specified in/etc/login.defs.

Reply:
If you're on an OS that does not report the `login_defs` data structure (like Windows), then you need to ensure that this gets a valid reported value and does not fail your compile.